tests/unit_tests/test_wave_07.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.item import Item
+
+#@pytest.mark.skip
+def test_swap_items_by_newest_items_returns_true():
+    item_a = Item(age=13)
+    item_b = Item(age=2)
+    item_c = Item(age=24)
+    fatimah = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    item_d = Item(age=10)
+    item_e = Item(age=14)
+    jolie = Vendor(
+        inventory=[item_d, item_e]
+    )
+
+    result = fatimah.swap_by_newest(jolie)
+
+    assert len(fatimah.inventory) == 3
+    assert item_b not in fatimah.inventory
+    assert item_a in fatimah.inventory
+    assert item_c in fatimah.inventory
+    assert item_d in fatimah.inventory
+    assert len(jolie.inventory) == 2
+    assert item_d not in jolie.inventory
+    assert item_e in jolie.inventory
+    assert item_b in jolie.inventory
+    assert result == True
+
+#@pytest.mark.skip
+def test_swap_items_when_my_items_age_is_missing_returns_false():
+    item_a = Item(age=50)
+    item_b = Item(age=14)
+    item_c = Item(age=8)
+    fatimah = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    item_d = Item(age=34)
+    item_e = Item(age=0)
+    jolie = Vendor(
+        inventory=[item_d, item_e]
+    )
+
+    result = fatimah.swap_by_newest(jolie)
+
+    assert len(fatimah.inventory) == 3
+    assert item_c not in fatimah.inventory
+    assert item_e in fatimah.inventory
+    assert item_b in fatimah.inventory
+    assert item_a in fatimah.inventory
+    assert len(jolie.inventory) == 2
+    assert item_d in jolie.inventory
+    assert item_e not in jolie.inventory
+    assert result == True
+
+#@pytest.mark.skip
+def test_swap_newest_items_when_is_missing_returns_false():
+    fatimah = Vendor(
+        inventory=[]
+        )
+
+    item_d = Item()
+    item_e = Item()
+    jolie = Vendor(
+        inventory=[item_d, item_e]
+    )
+
+    result = fatimah.swap_by_newest(jolie)
+
+    assert len(fatimah.inventory) == 0
+    assert item_d in jolie.inventory
+    assert item_e in jolie.inventory
+    assert len(jolie.inventory) == 2
+    assert item_d not in fatimah.inventory
+    assert item_e not in fatimah.inventory
+    assert result == False
+
+# @pytest.mark.skip
+def test_swap_newest_items_from_my_empty_returns_false():
+    fatimah = Vendor(
+        inventory=[]
+    )
+
+    item_d = Item()
+    item_e = Item()
+    jolie = Vendor(
+        inventory=[item_d, item_e]
+    )
+
+    result = fatimah.swap_by_newest(jolie)
+
+    with pytest.raises(ValueError):
+        if result == False:
+            raise ValueError(f"An exception occurred.")
```

Comment:
When you use `pytest.raises()` in a unit test, the function under test (`swap_by_newest`), should be in the codeblock that follows `pytest.raises()`. 

You invoke the method then check if `result == False` which does not rely on the tool to check that the method you wrote is throwing an error. You're checking if some value is equal to something else and then manually raising an error in the test on line 98.

When writing a test to see if the function under test is correctly raising an error for given invalid inputs, we should **not** be manually raising errors in tests. This is forcing the test to pass by saying "ok I need an error, I'm gonna make an error happen in the test by raising it". What we should be asking ourselves is "Will my function raise the error I think it does when I pass it specific inputs?"

This test should look like 
```py
def test_swap_newest_items_from_my_empty_returns_false():
    fatimah = Vendor(
        inventory=[]
    )

    item_d = Item()
    item_e = Item()
    jolie = Vendor(
        inventory=[item_d, item_e]
    )

    with pytest.raises(ValueError):
        fatimah.swap_by_newest(jolie)
```

Review these resources for more details: https://docs.pytest.org/en/stable/how-to/assert.html#assertraises and https://learn-2.galvanize.com/cohorts/4211/blocks/1032/content_files/exception-handling/checking-exceptions-in-tests.md